stellargraph/data/explorer.py

```diff
@@ -789,3 +791,152 @@ def _check_neighbourhood_sizes(self, in_size, out_size):
             self._raise_error(
                 "The number of hops for the in and out neighbourhoods must be the same."
             )
+
+
+@experimental(reason="requires more thorough testing and documentation")
+class TemporalRandomWalk(GraphWalk):
+    """
+    Performs uniform temporal random walks on the given graph
+    """
+
+    def run(
+        self,
+        nodes=None,
```

Comment:
Given #704, I imagine some of these should become required? (Just a reminder for this to be updated, depending on the merge order.)